codebuild/bin/s2n_setup_env.sh

```diff
@@ -151,7 +154,7 @@ fi
 export LIBFUZZER_ROOT=$LIBFUZZER_INSTALL_DIR
 
 #check if the path contains test dep X, if not and X exists, add to path
-path_overrides="$AWSLC_INSTALL_DIR/bin
+path_overrides="$LIBCRYPTO_ROOT/bin
```

Comment:
Can we just remove this line, rather than replace it with `LIBCRYPTO_ROOT`? Why do we need the `bin` directory of the libcrypto root in the path (outside of integration tests), especially since this wasn't needed before?

Would this avoid the new `LIBCRYPTO_ROOT` definitions for AL2? Or are these needed for some other reason?

Reply:
There are, with this PR, 5 possible [`bssl`](https://github.com/aws/s2n-tls/blob/main/tests/integrationv2/providers.py#L704) binaries available. If it's not in the `PATH`, we'd need to manually call out where to find it for the integration tests needing a BoringSSL provider.

Reply:
Ahh ok so this exists for the integration tests? Wouldn't the integration tests always expect to be able to find `bssl` though? Like, with this change, wouldn't an integration test with `bssl` only work when the s2n-tls libcrypto is also aws-lc?

Reply:
Reverted after offline discussion; I've also added some comments so future me doesn't try to repeat this change.